metrics-facade-base/src/main/java/com/ringcentral/platform/metrics/instrument/executors/AbstractMonitoredExecutorServiceBuilder.java

```diff
@@ -0,0 +1,176 @@
+package com.ringcentral.platform.metrics.instrument.executors;
+
+import com.ringcentral.platform.metrics.MetricKey;
+import com.ringcentral.platform.metrics.MetricRegistry;
+import com.ringcentral.platform.metrics.labels.Label;
+import com.ringcentral.platform.metrics.labels.LabelValues;
+import com.ringcentral.platform.metrics.names.MetricName;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+
+import static com.ringcentral.platform.metrics.PrefixLabelValuesMetricKey.prefixLabelValuesMetricKey;
+import static com.ringcentral.platform.metrics.labels.LabelValues.*;
+import static com.ringcentral.platform.metrics.names.MetricName.metricName;
+import static com.ringcentral.platform.metrics.utils.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public abstract class AbstractMonitoredExecutorServiceBuilder<ES extends ExecutorService, B extends AbstractMonitoredExecutorServiceBuilder<ES, B>> {
+
+    public static final MetricName DEFAULT_METRIC_NAME_PREFIX = metricName("executor", "service");
+    public static final Label NAME = new Label("name");
+
+    private final ES parent;
+    private final MetricRegistry registry;
+    private String name;
+    private boolean nameAsLabel = true;
+    private MetricName metricNamePrefix = DEFAULT_METRIC_NAME_PREFIX;
+    private LabelValues prefixLabelValues;
+    private LabelValues additionalLabelValues = noLabelValues();
+
+    protected static final AtomicLong orderNum = new AtomicLong();
+
+    /**
+     * Returns the next order number as a string (e.g., "1", "2", ...).
+     * This is used to generate a metric name when no explicit name is provided.
+     *
+     * @return the next order number as a string
+     */
+    protected static String nextOrderNumAsString() {
+        return Long.toString(orderNum.incrementAndGet());
+    }
+
+    /**
+     * @param parent   the parent {@link ExecutorService} that is being monitored
+     * @param registry the {@link MetricRegistry} used to register metrics
+     */
+    public AbstractMonitoredExecutorServiceBuilder(@Nonnull ES parent, @Nonnull MetricRegistry registry) {
+        this.parent = parent;
+        this.registry = registry;
+    }
+
+    /**
+     * Sets the name of the executor service.
+     * The name will either be included as part of the metric names (when {@code nameAsLabel(false)} is used),
```

Comment:
I suggest to replace detailed description with `@see` with link to `nameAsLabel` method.

Reply:
Replaced, thanks!